### PR TITLE
Added a check for whether the data directory is writable

### PR DESCRIPTION
**Pull request by robbestad**
Added a check for whether the data directory is writable, and provide a more exact error text in this case.

### CODE REVIEW

File: config/module.config.php [hunk 1]
```diff
@@ -50,7 +50,15 @@
 
 $inspectionsDir =  'data/roave_developer_tools';
 
-@mkdir($inspectionsDir, 0777, true);
+if(!is_dir($inspectionsDir)) {
+    try {
+        @mkdir($inspectionsDir, 0777, true);
+    }
+    catch (Exception $e){
```
Comment (Ocramius) [inline config/module.config.php]: No exceptions are raised during `mkdir`

File: src/Roave/DeveloperTools/Repository/Exception/InvalidFilePathException.php [hunk 2]
```diff
@@ -50,6 +50,11 @@ public static function fromUnReadableFile($path)
      */
     public static function fromUnWritableFile($path)
     {
+        if (! is_writable(dirname($path))) {
+            return new self(
+                sprintf('Invalid path provided, the path "%s" does not appear to be writable', $path)
+            );
+        }
```
Comment (Ocramius) [inline src/Roave/DeveloperTools/Repository/Exception/InvalidFilePathException.php:57]: Newline after this line

File: src/Roave/DeveloperTools/Repository/Exception/InvalidFilePathException.php [hunk 3]
```diff
@@ -50,6 +50,11 @@ public static function fromUnReadableFile($path)
...
         if (! file_exists(dirname($path))) {
```
Comment (Ocramius) [inline src/Roave/DeveloperTools/Repository/Exception/InvalidFilePathException.php:58]: This check should come before `is_writable`

File: config/module.config.php (hunk 1 above)
Comment (cordoval) [inline config/module.config.php]: true

File: config/module.config.php [hunk 4]
```diff
@@ -50,7 +50,15 @@
...
+        echo $e;
```
Comment (robbestad) [inline config/module.config.php]: Agreed. I'll add this now! However, as far as I could tell, the catch is not executed. EDIT: This should have given me a clue earlier. It's wrong to use try-catch here. I took it away.

File: config/module.config.php (hunk 1 above)
Comment (robbestad) [inline config/module.config.php]: Thoughts on how to handle it better?

Comment (Ocramius) [inline config/module.config.php]: @svenanders `Zend\Stdlib\ErrorHandler` maybe...

**Pull request closed without merging**